### PR TITLE
Feature/issue 75 develop a page for creating new product tour items

### PR DESCRIPTION
**Pull request by SimerdeepSinghGrewal**
added all the components asked in the issue.
added product tour page as well
changed TourList and TourListItem component to List and ListItem component as Vlad told me to make it generalized
Screenshot 2024-07-21 133304

Left menu also works and displays as required

## Summary by CodeRabbit

- **New Features**
  - Introduced a new route for the Product Tour, accessible via `/tours`.
  - Added new components: `ConfirmationPopup`, `ContentArea`, `ContentHeader`, `InfoTooltip`, `List`, and `ListItem` to enhance UI interactivity and organization.
  - Implemented a `TourDescriptionText` component for displaying descriptions.
  
- **Bug Fixes**
  - Improved readability and structure in the `Home` component without changing functionality.

- **Style**
  - New CSS styles for `ProductTour`, `ListItem`, and overall layout enhancements to improve visual presentation and user experience.

### CODE REVIEW

**Review by uparkalau — DISMISSED** [review]

File: frontend/src/components/LeftMenu/LeftMenu.jsx [hunk 1]
```diff
@@ -18,6 +18,9 @@ function LeftMenu() {
   const handleHomeClick = () => {
     navigate('/');
   };
+  const handleToursClick = () => {
+    navigate(-1);
```
Comment (erenfn) [inline frontend/src/components/LeftMenu/LeftMenu.jsx:22]: Doesn't this navigate to the previous page?

Comment (SimerdeepSinghGrewal) [reply frontend/src/components/LeftMenu/LeftMenu.jsx:22]: Tunc asked me to do that, like first we tried on my system lil bit n then he done everything on his system

Comment (SimerdeepSinghGrewal) [reply frontend/src/components/LeftMenu/LeftMenu.jsx:22]: I tried to "navigate('/tours');" do this and push it but my push was rejected.

**Review by uparkalau — APPROVED** [review]

File: frontend/src/components/List/ListItem/ListItem.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import SettingsIcon from '@mui/icons-material/Settings';
+import './ListItem.css';
+
+const ListItem = ({ title, text, id, onClick, onDelete, onEdit }) => {
+  return (
+    <div className="list-item" onClick={onClick}>
+      <div className="list-item-info">
+        <div className="list-item-header">
+          <svg width="16" height="16" viewBox="0 0 6 6" fill="none" xmlns="http://www.w3.org/2000/svg" className="list-item-icon">
+            <path d="M0 3C0 1.34315 1.34315 0 3 0V0C4.65685 0 6 1.34315 6 3V3C6 4.65685 4.65685 6 3 6V6C1.34315 6 0 4.65685 0 3V3Z" fill="#7F56D9"/>
```
Comment (uparkalau) [inline frontend/src/components/List/ListItem/ListItem.jsx:14]: what is this numbers, refactor it . what is svg from w3.org? change fill with variable from theme

File: frontend/src/scenes/productTour/ProductTourStyles.css [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  margin: 10px 0;
+  border: 1px solid transparent; /* Default to no border */
+  border-radius: 10px;
+  background-color: #FAFAFA;
+  cursor: pointer;
+}
+
+.list-item:hover,
+.list-item:focus {
+  border: 1px solid #7F56D9; /* Show border on hover or focus */
```
Comment (uparkalau) [inline frontend/src/scenes/productTour/ProductTourStyles.css:28]: use variables

File: frontend/src/scenes/productTour/ProductTourStyles.css [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+.list-item-info h4 {
+  margin: 0;
+  color: #344054;
+}
+
+.list-item-info p {
+  margin: 0;
+  color: #667085;
```
Comment (uparkalau) [inline frontend/src/scenes/productTour/ProductTourStyles.css:54]: use variables for colors

**Review by uparkalau — CHANGES_REQUESTED** [review]
please resolve all comments before merge

**Review by SimerdeepSinghGrewal — COMMENTED** [review]
changed svg icon with mui icon and used variables for  colors

**Comment by uparkalau** [comment]
@coderabbitai review

File: frontend/src/components/TourDescriptionText/TourDescriptionText.jsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const TourDescriptionText = ({ description }) => {
+  return <p className="tour-description-text">{description}</p>;
+};
+
+TourDescriptionText.propTypes = {
+  description: PropTypes.string,
+};
```
[the remark this answers is not in the document]
Comment (SimerdeepSinghGrewal) [reply frontend/src/components/TourDescriptionText/TourDescriptionText.jsx:10]: I was asked to remove "isRequired"

File: frontend/src/components/InfoTooltip/InfoTooltip.jsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+      <IconButton>
+        <InfoIcon />
+      </IconButton>
+    </Tooltip>
+  );
+};
+
+InfoTooltip.propTypes = {
+  text: PropTypes.string,
+  title: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (SimerdeepSinghGrewal) [reply frontend/src/components/InfoTooltip/InfoTooltip.jsx:18]: I was asked to remove "isRequired"

File: frontend/src/components/InfoTooltip/InfoTooltip.jsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+};
```
[the remark this answers is not in the document]
Comment (SimerdeepSinghGrewal) [reply frontend/src/components/InfoTooltip/InfoTooltip.jsx:19]: I was told to add both in the issuse so added both

File: frontend/src/components/List/List.jsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+          onEdit={item.onEdit}
+        />
+      ))}
+    </div>
+  );
+};
+
+List.propTypes = {
+  items: PropTypes.arrayOf(PropTypes.object),
+  onSelectItem: PropTypes.func,
```
[the remark this answers is not in the document]
Comment (SimerdeepSinghGrewal) [reply frontend/src/components/List/List.jsx:25]: I was asked to remove "isRequired"

File: frontend/src/components/List/List.jsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+};
```
[the remark this answers is not in the document]
Comment (SimerdeepSinghGrewal) [reply frontend/src/components/List/List.jsx:26]: I was asked to remove "isRequired"

**Review by uparkalau — COMMENTED** [review]
@coderabbitai review this PR

**Review by uparkalau — DISMISSED** [review]

**Comment by erenfn** [comment]
Do not merge it yet

**Comment by SimerdeepSinghGrewal** [comment]
upon clicking "Create a tour" button this page comes
Screenshot 2024-08-09 141249

**Review by erenfn — DISMISSED** [review]

**Review by erenfn — APPROVED** [review]

**Pull request merged**